craft_store/base_client.py

```diff
@@ -32,6 +32,43 @@
 logger = logging.getLogger(__name__)
 
 
+def wrap_credentials(token_type: str, credentials: str) -> str:
+    """Create a payload string that contains both `credentials` and its identifying type.
+
+    :param token_type: The identifier for the type of token stored in `credentials`
+    :param credentials:
+    :return: A payload string ready to be passed to Auth.set_credentials()
+    """
+    return json.dumps({"t": token_type, "v": credentials})
+
+
+def unwrap_credentials(token_type: str, stored_credentials: str) -> str:
+    """Retrieve the type-specific "inner" credentials from `credentials`.
+
+    This function also handles backwards-compatibility by supporting `credentials`
+    from before we stored the `token_type`; it is meant to be called with the
+    returned value from Auth.get_credentials().
+
+    :param token_type: The expected token type, for validation.
+    :param stored_credentials: The credentials retrieved from auth storage.
+    """
+    try:
+        loaded = json.loads(stored_credentials)
+    except json.JSONDecodeError:
+        # Not a JSON string, so this is already the desired value
+        return stored_credentials
+
+    if "t" in loaded:
+        if loaded["t"] == token_type:
+            return loaded["v"]
+        # TODO: What should we do in the (buggy) case where the token type is different
```

Comment:
any suggestions on what to do here? Is there maybe a better error type?

Reply:
as I was looking at the code I was thinking that it might be a good idea to have the credentials encapsulated in a pydantic model (two, one for each) that would marshal/unmarshal accordingly

Reply:
this however does not answer your question

Reply:
Haven't tried the Pydantic suggestion yet (should we)? But I left as a `CredentialsNotParseable` because it seems right "enough". Either that or maybe we can add a `CredentialsNotSupported` error

Reply:
Left as CredentialsNotParseable (with a better message), added tests for this